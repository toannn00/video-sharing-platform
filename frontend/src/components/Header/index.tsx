import { Row, Col, Divider } from "antd";
import { Login } from "../Login";
import { Link } from "react-router-dom";

export const Header = () => {
  return (
    <>
      <Row align="middle">
        <Col span={9} offset={3}>
          <Link to="/">
            <div style={{ display: "flex" }}>
              <h1>Video App</h1>
            </div>
          </Link>
        </Col>
        <Col span={9} offset={1}>
          <Login />
        </Col>
      </Row>
      <Divider style={{ marginTop: 5 }} />
    </>
  );
};

export default Header;
