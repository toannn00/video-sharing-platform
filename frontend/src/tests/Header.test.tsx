import "@testing-library/jest-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import Header from "../components/Header";

describe("Header", () => {
  test("should render and match the snapshot", () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<Header />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
