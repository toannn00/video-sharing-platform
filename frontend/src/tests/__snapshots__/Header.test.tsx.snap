// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should render and match the snapshot 1`] = `
<React.Fragment>
  <Row
    align="middle"
    gutter={
      [
        0,
        0,
      ]
    }
    style={
      {
        "flexDirection": "row",
        "padding": "16px 0",
      }
    }
  >
    <Col
      md={
        {
          "offset": 3,
          "span": 9,
        }
      }
      style={
        {
          "display": "flex",
          "justifyContent": "flex-start",
        }
      }
      xs={24}
    >
      <Link
        to="/"
      >
        <div
          style={
            {
              "display": "flex",
            }
          }
        >
          <LogoIcon />
        </div>
      </Link>
    </Col>
    <Col
      md={
        {
          "offset": 1,
          "span": 9,
        }
      }
      style={
        {
          "display": "flex",
          "justifyContent": "flex-start",
        }
      }
      xs={24}
    >
      <Login />
    </Col>
  </Row>
  <Divider
    style={
      {
        "margin": "5px 0 0 0",
      }
    }
  />
</React.Fragment>
`;
